### PR TITLE
[SDK-614] Add new Search Parameter

### PR DESCRIPTION
**Pull request by bitPogo**
## Description

This adds the new search parameter in order to align the capabilities of the SDK with its JS sibling and to satisfy the new BackendApi. This also alters Meta to expose the RecordStatus.

## Motivation and Context

The Backend introduced new query parameter (start_updated_date, end_updated_date and include_deleted), which are usable with this PR. For more see the [pending ticket](https://gesundheitscloud.atlassian.net/browse/SDK-612).

## How is it being implemented?

In order to provide the new parameter mainly the SdkDateFormatter had been altered and is not longer responsible for building the Meta. The Formatter has now an extended Interface for formatting/parsing LocalDate and LocalDateTime alike.
In order to make the Parameter useable 2 new Parameter classes had been introduced (CreationDateRange and UpdateDateTimeRange) and wired in to the public Api.
Any logic in regards to resolving the ranges had been pulled out the RecordService and moved to a separated class (DateResolver).
The ApiService and the HealthCloudApi had been extended in order to make actual calls.

In regards to the RecordStatus a new Enum had been introduced and the Meta got extended. Also the RecordDataMapper and RecordEncryptionService had been altered to deal with the changes accordingly.
Lastly a new TypeAdapter was needed to serialise the RecordStatus.

## How Has This Been Tested?

The tests sets for search/fetch had been splitted to separate fetch and search from each other (unit test and module test).

All effected classes had been altered/extended by unit tests to cover the changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by wmontwe** [comment]
@bitPogo this has a conflict with main branch

File: sdk-core/src/test/java/care/data4life/sdk/wrapper/URLEncodingTest.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ import org.junit.Test
 import kotlin.test.assertEquals
 import kotlin.test.assertTrue
 
-class UrlEncodingTest {
+class URLEncodingTest {
```
Comment (wmontwe) [inline sdk-core/src/test/java/care/data4life/sdk/wrapper/URLEncodingTest.kt:23]: This should be lowercase, abreviations are also camel cased. This is easier to read.

Comment (bitPogo) [reply sdk-core/src/test/java/care/data4life/sdk/wrapper/URLEncodingTest.kt]: Done

Comment (bitPogo) [reply sdk-core/src/test/java/care/data4life/sdk/wrapper/URLEncodingTest.kt]: done

**Pull request merged**